resources/patches/ruby_241/PA-1124_add_nano_server_com_support-8feb9779182bd4285f3881029fe850dac188c1ac.patch

```diff
@@ -0,0 +1,91 @@
+From 8feb9779182bd4285f3881029fe850dac188c1ac Mon Sep 17 00:00:00 2001
+From: suke <suke@b2dd03c8-39d4-4d8f-98ff-823fe69b080e>
+Date: Sat, 25 Feb 2017 22:28:22 +0000
+Subject: [PATCH] ext/win32ole/win32ole.c(ole_initialize): avoid to fail in
+ Windows nano server. This is experimental. Thanks to mwrock, Ethan Brown.
+
+git-svn-id: svn+ssh://ci.ruby-lang.org/ruby/trunk@57715 b2dd03c8-39d4-4d8f-98ff-823fe69b080e
+---
+ ext/win32ole/win32ole.c | 37 +++++++++++++++++++++++++++++++++----
+ 1 file changed, 33 insertions(+), 4 deletions(-)
+
+diff --git a/ext/win32ole/win32ole.c b/ext/win32ole/win32ole.c
+index 9518e2f0cd91..b05ace71aeb5 100644
+--- a/ext/win32ole/win32ole.c
++++ b/ext/win32ole/win32ole.c
+@@ -50,6 +50,7 @@ static volatile DWORD g_ole_initialized_key = TLS_OUT_OF_INDEXES;
+ static BOOL g_uninitialize_hooked = FALSE;
+ static BOOL g_cp_installed = FALSE;
+ static BOOL g_lcid_installed = FALSE;
++static BOOL g_running_nano = FALSE;
+ static HINSTANCE ghhctrl = NULL;
+ static HINSTANCE gole32 = NULL;
+ static FNCOCREATEINSTANCEEX *gCoCreateInstanceEx = NULL;
+@@ -169,6 +170,7 @@ static VALUE fole_activex_initialize(VALUE self);
+ static void com_hash_free(void *ptr);
+ static void com_hash_mark(void *ptr);
+ static size_t com_hash_size(const void *ptr);
++static void check_nano_server(void);
+ 
+ static const rb_data_type_t ole_datatype = {
+     "win32ole",
+@@ -817,16 +819,23 @@ ole_initialize(void)
+     }
+ 
+     if(g_ole_initialized == FALSE) {
++        if(g_running_nano) {
++            hr = CoInitializeEx(NULL, COINIT_MULTITHREADED);
++        } else {
++            hr = OleInitialize(NULL);
++        }
+         hr = OleInitialize(NULL);
```

Comment:
Looks like this was accidentally left in - meaning the hresult is always from OleInitialize instead of from the conditional. Possibly misapplication of the original patch, as its not in https://github.com/ruby/ruby/pull/1423